<a href="https://colab.research.google.com/github/a-taylor1/DATA450_Team5/blob/main/Assignment11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from mlxtend.frequent_patterns import apriori
from mlxtend.frequent_patterns import association_rules

import warnings
warnings.filterwarnings('ignore', category=DeprecationWarning)

%matplotlib inline

In [ ]:
data = pd.read_csv("https://raw.githubusercontent.com/eitellauria/datafiles/main/ShoppingData.csv")

In [ ]:
itemsets = apriori(data, min_support=0.1, use_colnames=True)
itemsets

,support,itemsets
0,0.492366,(ReadyMade)
1,0.402036,(FrozenFoods)
2,0.394402,(Alcohol)
3,0.188295,(Milk)
4,0.428753,(BakeryGoods)
5,0.474555,(Snacks)
6,0.455471,(TinnedGoods)
7,0.211196,"(ReadyMade, FrozenFoods)"
8,0.212468,"(Alcohol, ReadyMade)"
9,0.133588,"(ReadyMade, Milk)"


In [ ]:
rules = association_rules(itemsets, metric='confidence', min_threshold=0.8)
rules = rules[[len(c) <= 3 for c in rules.antecedents]]
rules = rules[[c >= 1.5 for c in rules.lift]]
rules.drop(columns=['antecedent support', 'consequent support', 'conviction', 'leverage', 'zhangs_metric', 'representativity','jaccard','certainty', 'kulczynski'])

,antecedents,consequents,support,confidence,lift
0,"(Alcohol, BakeryGoods, TinnedGoods)",(ReadyMade),0.100509,0.814433,1.654120
1,"(Alcohol, ReadyMade, TinnedGoods)",(BakeryGoods),0.100509,0.831579,1.939528


1. If a customer purchases alcohol, baked goods, and tinned goods, we are 81% confident that they will also purchase Ready Made.
2. If a customer purchase alcohol, Ready Made, and tinned goods, we are 83% confident that they will also purchase baked goods.